volatility3/framework/plugins/linux/library_list.py

```diff
@@ -0,0 +1,169 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import Iterable, Tuple
+
+from volatility3.framework import interfaces, renderers, constants, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.objects import utility
+from volatility3.framework.symbols import intermed
+from volatility3.framework.symbols.linux.extensions import elf
+from volatility3.plugins.linux import pslist
+
+
+vollog = logging.getLogger(__name__)
+
+
+class LibraryList(interfaces.plugins.PluginInterface):
+    """Enumerate libraries loaded into processes"""
+
+    _required_framework_version = (2, 0, 0)
+
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 2, 0)
+            ),
+            requirements.ListRequirement(
+                name="pids",
+                description="Filter on specific process IDs",
+                element_type=int,
+                optional=True,
+            ),
+        ]
+
+    def get_libdl_libraries(
```

Comment:
Did you want to expose this as a class method for other plugins to be able to use it?  If so the `context` and `config_path` need parameterizing (usually as the first and second parameters), but otherwise it doesn't look like it'll need anything special...

Reply:
It's not intended to be a class method

Reply:
Ok, then *possibly* better to start it with an `_` to show it's supposed to be private?  Just a style thing, but it's why I wasn't immediately sure...